internal/tools/aws/client.go

```diff
@@ -73,6 +61,45 @@ type Client struct {
 	secretsManager secretsmanageriface.SecretsManagerAPI
 }
 
+// NewService creates a new instance of Service.
+func NewService(sess *session.Session) *Service {
+	return &Service{
+		acm:            acm.New(sess),
+		iam:            iam.New(sess),
+		rds:            rds.New(sess),
+		s3:             s3.New(sess),
+		route53:        route53.New(sess),
+		secretsManager: secretsmanager.New(sess),
+		ec2:            ec2.New(sess),
+	}
+}
+
+// Client is a client for interacting with AWS resources.
+type Client struct {
+	store   model.InstallationDatabaseStoreInterface
+	service *Service
+	config  *aws.Config
+	mux     *sync.Mutex
+}
+
+// Service contructs an AWS session if not yet successfully done and returns AWS clients.
+func (c *Client) Service() *Service {
+	if c.service == nil {
+		sess, err := NewAWSSessionWithLogger(c.config, log.WithField("tools-aws", "client"))
```

Comment:
Does it makes sense to pass in a logger like we do most other places?

Reply:
👍